File: src/scripts/models/project.ts
import Base from './base';

class Project implements Base {
  id: string;

  fileName: string;

  fileType: string;

  createdAt: Date;

  createdBy: string;

  modifiedAt: Date;

  modifiedBy: string;

  path: string = '/';

  constructor(
    id: string,
    fileName: string,
    fileType: string,
    path: string = '/',
    createdAt: Date,
    createdBy: string,
    modifiedAt: Date,
    modifiedBy: string,
  ) {
    this.id = id;
    this.fileName = fileName;
    this.fileType = fileType;
    this.createdAt = createdAt;
    this.createdBy = createdBy;
    this.modifiedAt = modifiedAt;
    this.modifiedBy = modifiedBy;
    this.path = path;
  }
}

export default Project;

File: Models/Project.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace backend.Models
{
    public class Project
    {
        public int Id { get; set; }

        public string FileName { get; set; }

        public DateTime CreatedAt { get; set; }

        public string CreatedBy { get; set; }

        public DateTime ModifiedAt { get; set; }

        public string ModifiedBy { get; set; }
    }
}
